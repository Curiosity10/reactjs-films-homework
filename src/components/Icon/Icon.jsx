import React from 'react';
import PropTypes from 'prop-types';

const Icon = (props) => {
  const { name, color } = props;
  switch (name) {
    case 'search':
      return (
        <svg fill={color} width="15" height="20" x="0px" y="0px" viewBox="0 0 56.966 56.966">
          <path d="M55.146,51.887L41.588,37.786c3.486-4.144,5.396-9.358,5.396-14.786c0-12.682-10.318-23-23-23s-23,10.318-23,23
            s10.318,23,23,23c4.761,0,9.298-1.436,13.177-4.162l13.661,14.208c0.571,0.593,1.339,0.92,2.162,0.92
            c0.779,0,1.518-0.297,2.079-0.837C56.255,54.982,56.293,53.08,55.146,51.887z M23.984,6c9.374,0,17,7.626,17,17s-7.626,17-17,17
            s-17-7.626-17-17S14.61,6,23.984,6z"
          />
        </svg>
      );
    case 'grid':
      return (
        <svg width="20" height="20" x="0px" y="0px" viewBox="0 0 271.673 271.673" fill={color}>
          <g>
            <g>
              <g>
                <path d="M114.939,0H10.449C4.678,0,0,4.678,0,10.449v104.49c0,5.771,4.678,10.449,10.449,10.449h104.49
                  c5.771,0,10.449-4.678,10.449-10.449V10.449C125.388,4.678,120.71,0,114.939,0z"
                />
                <path d="M261.224,0h-104.49c-5.771,0-10.449,4.678-10.449,10.449v104.49c0,5.771,4.678,10.449,10.449,10.449h104.49
                  c5.771,0,10.449-4.678,10.449-10.449V10.449C271.673,4.678,266.995,0,261.224,0z"
                />
                <path d="M114.939,146.286H10.449C4.678,146.286,0,150.964,0,156.735v104.49c0,5.771,4.678,10.449,10.449,10.449h104.49
                  c5.771,0,10.449-4.678,10.449-10.449v-104.49C125.388,150.964,120.71,146.286,114.939,146.286z"
                />
                <path d="M261.224,146.286h-104.49c-5.771,0-10.449,4.678-10.449,10.449v104.49c0,5.771,4.678,10.449,10.449,10.449h104.49
                  c5.771,0,10.449-4.678,10.449-10.449v-104.49C271.673,150.964,266.995,146.286,261.224,146.286z"
                />
              </g>
            </g>
          </g>
        </svg>
      );
    case 'table':
      return (
        <svg fill={color} width="20" height="20" x="0px" y="0px" viewBox="0 0 512 512">
          <g>
            <g>
              <g>
                <rect width="100%" height="235" />
                <rect y="277.333" width="100%" height="235" />
              </g>
            </g>
          </g>
        </svg>
      );
    case 'play':
      return (
        <svg width="80px" height="80px" x="0px" y="0px" viewBox="0 0 58 58">
          <circle style={{ fill: color }} cx="29" cy="29" r="29" />
          <g>
            <polygon style={{ fill: '#fff' }} points="44,29 22,44 22,29.273 22,14" />
            <path
              style={{ fill: '#fff' }}
              d="M22,45c-0.16,0-0.321-0.038-0.467-0.116C21.205,44.711,21,44.371,21,44V14
            c0-0.371,0.205-0.711,0.533-0.884c0.328-0.174,0.724-0.15,1.031,0.058l22,15C44.836,28.36,45,28.669,45,29s-0.164,0.64-0.437,0.826
            l-22,15C22.394,44.941,22.197,45,22,45z M23,15.893v26.215L42.225,29L23,15.893z"
            />
          </g>
        </svg>
      );
    case 'close':
      return (
        <svg width="15" height="15" viewBox="0 0 64 64">
          <g>
            <path
              fill={color}
              d="M28.941,31.786L0.613,60.114c-0.787,0.787-0.787,2.062,0,2.849c0.393,0.394,0.909,0.59,1.424,0.59   c0.516,0,1.031-0.196,1.424-0.59l28.541-28.541l28.541,28.541c0.394,0.394,0.909,0.59,1.424,0.59c0.515,0,1.031-0.196,1.424-0.59   c0.787-0.787,0.787-2.062,0-2.849L35.064,31.786L63.41,3.438c0.787-0.787,0.787-2.062,0-2.849c-0.787-0.786-2.062-0.786-2.848,0   L32.003,29.15L3.441,0.59c-0.787-0.786-2.061-0.786-2.848,0c-0.787,0.787-0.787,2.062,0,2.849L28.941,31.786z"
            />
          </g>
        </svg>
      );
    default:
      return null;
  }
};

Icon.propTypes = {
  name: PropTypes.string.isRequired,
  color: PropTypes.string.isRequired,
};

export default Icon;
