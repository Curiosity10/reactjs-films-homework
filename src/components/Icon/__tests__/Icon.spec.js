import React from 'react';
import { create } from 'react-test-renderer';
import Icon from '..';

describe('Icon component renders correctly', () => {
  it('Search icon renders correctly', () => {
    const tree = create(<Icon color="#fff" name="search" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Grid icon renders correctly', () => {
    const tree = create(<Icon color="#fff" name="grid" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Table icon renders correctly', () => {
    const tree = create(<Icon color="#fff" name="table" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Play icon renders correctly', () => {
    const tree = create(<Icon color="#fff" name="play" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Close icon renders correctly', () => {
    const tree = create(<Icon color="#fff" name="close" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Default no icon', () => {
    const tree = create(<Icon color="#fff" name="" />).toJSON();
    expect(tree).toBe(null);
  });
});
