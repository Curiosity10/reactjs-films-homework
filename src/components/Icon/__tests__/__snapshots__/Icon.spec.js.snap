// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon component renders correctly Close icon renders correctly 1`] = `
<svg
  height="15"
  viewBox="0 0 64 64"
  width="15"
>
  <g>
    <path
      d="M28.941,31.786L0.613,60.114c-0.787,0.787-0.787,2.062,0,2.849c0.393,0.394,0.909,0.59,1.424,0.59   c0.516,0,1.031-0.196,1.424-0.59l28.541-28.541l28.541,28.541c0.394,0.394,0.909,0.59,1.424,0.59c0.515,0,1.031-0.196,1.424-0.59   c0.787-0.787,0.787-2.062,0-2.849L35.064,31.786L63.41,3.438c0.787-0.787,0.787-2.062,0-2.849c-0.787-0.786-2.062-0.786-2.848,0   L32.003,29.15L3.441,0.59c-0.787-0.786-2.061-0.786-2.848,0c-0.787,0.787-0.787,2.062,0,2.849L28.941,31.786z"
      fill="#fff"
    />
  </g>
</svg>
`;

exports[`Icon component renders correctly Grid icon renders correctly 1`] = `
<svg
  fill="#fff"
  height="20"
  viewBox="0 0 271.673 271.673"
  width="20"
  x="0px"
  y="0px"
>
  <g>
    <g>
      <g>
        <path
          d="M114.939,0H10.449C4.678,0,0,4.678,0,10.449v104.49c0,5.771,4.678,10.449,10.449,10.449h104.49 c5.771,0,10.449-4.678,10.449-10.449V10.449C125.388,4.678,120.71,0,114.939,0z"
        />
        <path
          d="M261.224,0h-104.49c-5.771,0-10.449,4.678-10.449,10.449v104.49c0,5.771,4.678,10.449,10.449,10.449h104.49 c5.771,0,10.449-4.678,10.449-10.449V10.449C271.673,4.678,266.995,0,261.224,0z"
        />
        <path
          d="M114.939,146.286H10.449C4.678,146.286,0,150.964,0,156.735v104.49c0,5.771,4.678,10.449,10.449,10.449h104.49 c5.771,0,10.449-4.678,10.449-10.449v-104.49C125.388,150.964,120.71,146.286,114.939,146.286z"
        />
        <path
          d="M261.224,146.286h-104.49c-5.771,0-10.449,4.678-10.449,10.449v104.49c0,5.771,4.678,10.449,10.449,10.449h104.49 c5.771,0,10.449-4.678,10.449-10.449v-104.49C271.673,150.964,266.995,146.286,261.224,146.286z"
        />
      </g>
    </g>
  </g>
</svg>
`;

exports[`Icon component renders correctly Play icon renders correctly 1`] = `
<svg
  height="80px"
  viewBox="0 0 58 58"
  width="80px"
  x="0px"
  y="0px"
>
  <circle
    cx="29"
    cy="29"
    r="29"
    style={
      Object {
        "fill": "#fff",
      }
    }
  />
  <g>
    <polygon
      points="44,29 22,44 22,29.273 22,14"
      style={
        Object {
          "fill": "#fff",
        }
      }
    />
    <path
      d="M22,45c-0.16,0-0.321-0.038-0.467-0.116C21.205,44.711,21,44.371,21,44V14 c0-0.371,0.205-0.711,0.533-0.884c0.328-0.174,0.724-0.15,1.031,0.058l22,15C44.836,28.36,45,28.669,45,29s-0.164,0.64-0.437,0.826 l-22,15C22.394,44.941,22.197,45,22,45z M23,15.893v26.215L42.225,29L23,15.893z"
      style={
        Object {
          "fill": "#fff",
        }
      }
    />
  </g>
</svg>
`;

exports[`Icon component renders correctly Search icon renders correctly 1`] = `
<svg
  fill="#fff"
  height="20"
  viewBox="0 0 56.966 56.966"
  width="15"
  x="0px"
  y="0px"
>
  <path
    d="M55.146,51.887L41.588,37.786c3.486-4.144,5.396-9.358,5.396-14.786c0-12.682-10.318-23-23-23s-23,10.318-23,23 s10.318,23,23,23c4.761,0,9.298-1.436,13.177-4.162l13.661,14.208c0.571,0.593,1.339,0.92,2.162,0.92 c0.779,0,1.518-0.297,2.079-0.837C56.255,54.982,56.293,53.08,55.146,51.887z M23.984,6c9.374,0,17,7.626,17,17s-7.626,17-17,17 s-17-7.626-17-17S14.61,6,23.984,6z"
  />
</svg>
`;

exports[`Icon component renders correctly Table icon renders correctly 1`] = `
<svg
  fill="#fff"
  height="20"
  viewBox="0 0 512 512"
  width="20"
  x="0px"
  y="0px"
>
  <g>
    <g>
      <g>
        <rect
          height="235"
          width="100%"
        />
        <rect
          height="235"
          width="100%"
          y="277.333"
        />
      </g>
    </g>
  </g>
</svg>
`;
